lib/shoryuken/options.rb

```diff
@@ -65,8 +65,8 @@ def polling_strategy(group)
         Polling::WeightedRoundRobin
       when 'StrictPriority'
         Polling::StrictPriority
-      when Class
-        strategy
+      when String
```

Comment:
@agrajgarg1526 would this change cause issues for users defining a `polling_strategy` with Ruby just passing a class?

Reply:
@phstc I am not aware of how can I pass a whole class to `polling_strategy` in yaml file configuration. But even for that case, we can add ` when String
      Object.const_get(strategy)` as a separate case above `when Class
        strategy`. This will not break anything. Thoughts?

Reply:
@agrajgarg1526 that's a good point. Can users can access `groups` and mutate it (adding `polling_strategy`)? If so, although unlikely, changing this can be a breaking change.

```ruby
group[...][:polling_strategy] = MyClass
``

Reply:
Sorry for being extremely careful. I just want to avoid breaking changes, if any.

Reply:
@phstc I tried doing `group[...][:polling_strategy] = MyClass` in`polling_strategy` function using monkey patching and it got matched with `Class`. But I don't think users can access `groups` and mutate it without overriding functions. But still to avoid this, keeping both the cases would be the best approach here imo.

Reply:
Thank you for double-checking this. I agree with keeping both.

Reply:
I'm running the test suite 🙏

Reply:
@agrajgarg1526 can you check the test errors?

https://github.com/ruby-shoryuken/shoryuken/actions/runs/7835447741/job/21380862789

Reply:
@phstc moved the argument error exception inside the `when String` case.